### PR TITLE
New documentation structure

### LINKED ISSUE #620: Documentation unfrendly for newbies
Quite often we get feedback that the reatom documentation is frustrating.
This is a significant problem that scares off new users, we need to try to be closer to what they need.
Next, I will present suggestions for improvement based on the feedback:

- [ ] "Introduction" (main page), with few words about what reatom is, list of features with links, simple as possible hello world example.   
- [ ] "Getting started" - this mostly about boilerplate actions and code that user probably need to do:
  - Setup - Covers installation and bindings.
    - Installation
    - TypeScript
    - ESlint
    - React binding
    - Solid binding
    - Vue binding
  - Learning - Provides a few examples that explain core concepts and the main flow.
    - atom
    - action
    - context
    - Style guide and best practices 
  - Debugging
  - Testing - Discusses testing strategies
    
 - [ ] "Recipes" - Offers short examples that solve common use cases.
   -  Share component state
      - in react
      - in solid
      - in vue
   - Data fetching
     - Load data when component mounted 
     - Handing loading state
     - Handing errors
     - Parallel request
     - Sequential requests
     - Concurrent requests with canceling
   - Forms
     - Text input
     - Radio
     - Checkbox
     - Select
     - Validation
   - Undo / Redo
   - Data persisting
- [ ] "Examples" - Includes links to repositories with examples and pet projects.
- [ ] "Handbook" - Provides an in-depth exploration of reactive programming and the underlying features of Reatom.
- [ ] "Packages"
  - [Package]: Offers the raw API documentation for the specific package.
- REPL
- Compat

### PR DESCRIPTION
**Pull request by Akiyamka**
fixes #620

### CODE REVIEW

File: docs/src/content/docs/getting-started/try-online.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+---
+title: Try online
+description: Links for online sandboxes
+sidebar: 
+  order: 0
+---
+
+
```
Comment (artalar) [inline docs/src/content/docs/getting-started/try-online.md:12]:
```suggestion
Here is the base template project with Vite, TypeScript, React and Reatom ecosystem: https://github.com/artalar/reatom-react-ts

Try it online: [codesandbox](https://codesandbox.io/p/sandbox/github/artalar/reatom-react-ts/tree/main), [stackblitz](https://githubblitz.com/artalar/reatom-react-ts), [gitpod](https://gitpod.io/#https://github.com/artalar/reatom-react-ts)

```

File: docs/src/content/docs/getting-started/setup.md [hunk 2]
```diff
@@ -0,0 +1,218 @@
+---
+title: Setup
+description: Explaining initial setup
+sidebar: 
+  order: 1
+---
+
```
Comment (artalar) [inline docs/src/content/docs/getting-started/setup.md:7]:
```suggestion

Reatom is a framework-agnostic state manager, and you can use it with various adapters for different frameworks. This guide provides a common usage with React.js, as it is the most commonly used view library currently.

```

File: docs/src/content/docs/getting-started/setup.md [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+```ts
+/* tsconfig: strictNullChecks: true */
+// AtomMut<string | null>
+const nullableAtom = atom<string | null>(null) 
+```
+
+You can play with this example on [typescript playground](https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBAbzgQwMY2BAdgGhTCEOAXzgDMpC4ByAASgFNkCQB6VaB6gKF46wDO8LAFcQAQRZwAvPkIAKAMwBKANx9sQuEKiSqs5gupkIEamo2D4GMHqIGW8+egAeymQD5E3OHH5bRez8YFwA6ATAAT3lAu2UfP014HRlgsIjonTiEhOAyOBixOC8ABncEBN9GGBEoLDhA9V9SBgAbAQZvXyqGGrrtGCgE4m5iC25-YRFW1uQAI1aGO1TDEAAeHWAsAHM4AB8G6daPQpm1IA)
+
+### With ESlint
```
Comment (artalar) [inline docs/src/content/docs/getting-started/setup.md]: MOVE IT TO SEPARATE GUIDLE PLEASE. Let's not add extra complexity for newbie right from the first steps.

File: docs/src/content/docs/getting-started/setup.md [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+
+export const App = () => (
+  <reatomContext.Provider value={ctx}>
+    <Main />
+  </reatomContext.Provider>
+)
+
+```
+
+React 16 and 17:
```
Comment (artalar) [inline docs/src/content/docs/getting-started/setup.md]: Definitely, there is excessive information. We should only focus on the main flow.

File: docs/src/content/docs/getting-started/setup.md [hunk 5]
```diff
@@ -0,0 +1,218 @@
...
+
+export const App = () => (
+  <reatomContext.Provider value={ctx}>
+    <Main />
+  </reatomContext.Provider>
+)
+
+```
+
+For `react-native`:
```
Comment (artalar) [inline docs/src/content/docs/getting-started/setup.md]: There's no need for it there too.

File: docs/src/content/docs/getting-started/setup.md [hunk 6]
```diff
@@ -0,0 +1,218 @@
...
+
+```jsx
+const nameAtom = atom('Joe')
+const handleNameChange = action(
+  (ctx, event) => nameAtom(ctx, event.currentTarget.value),
+  'handleNameChange',
+)
+
+const Greeting = () => {
+  const [name] = useAtom(nameAtom)
```
Comment (artalar) [inline docs/src/content/docs/getting-started/setup.md]: may be we will use `setName` here and don't mention `action` at all?

- less complexity
- `useAtom` will be more familiar and understandable with the setter, just like `useState`

File: docs/src/content/docs/getting-started/learning.md [hunk 7]
```diff
@@ -0,0 +1,177 @@
...
+```
+
+## Atoms are storing data and dependencies
+
+Let's have trivial example that employs reatom's reactivity by implementing simple `a + b = c`:
+
+```typescript
+import { createCtx, atom } from "@reatom/framework"
+
+const ctx = createCtx()
```
Comment (artalar) [inline docs/src/content/docs/getting-started/learning.md:23]: Move `ctx` creation bellow. It is important to focus on the context last of all. Add extra comment that `ctx` created only once.

File: docs/src/content/docs/getting-started/learning.md [hunk 8]
```diff
@@ -0,0 +1,177 @@
...
+const cAtom = atom((ctx) =>
+  ctx.spy(aAtom) + ctx.spy(bAtom),
+  'cAtom'
+)
+
+ctx.subscribe(cAtom, (c) => {
+  const a = ctx.get(aAtom);
+  const b = ctx.get(bAtom);
+  console.log(`${a} + ${b} = ${c}`)
+})
```
Comment (artalar) [inline docs/src/content/docs/getting-started/learning.md]: Computation in the `subscribe` function is what you should avoid initially. It should serve as a bridge for side effects only.

File: docs/src/content/docs/getting-started/learning.md [hunk 9]
```diff
@@ -0,0 +1,177 @@
...
+
+aAtom(ctx, 3)
+// logs: 3 + 2 = 5
+bAtom(ctx, 4)
+// logs: 3 + 4 = 7
+bAtom(ctx, 4)
+// does not log anything
+console.log(ctx.get(aAtom), ctx.get(bAtom), ctx.get(cAtom))
+// logs: 3, 4, 7
```
Comment (artalar) [inline docs/src/content/docs/getting-started/learning.md]:
```suggestion
aAtom(ctx, 3)
// logs: 3 + 2 = 5

bAtom(ctx, 4)
// logs: 3 + 4 = 7

bAtom(ctx, 4)
// does not log anything, as the state is not changed

console.log(ctx.get(aAtom), ctx.get(bAtom), ctx.get(cAtom))
// logs: 3, 4, 7
```

File: docs/src/content/docs/getting-started/learning.md [hunk 10]
```diff
@@ -0,0 +1,177 @@
...
+Moreover we can spy on any atom that enables us to make dependency trees like this:
+
+```typescript
+const cDoubledAtom = atom((ctx) => 2 * ctx.spy(cAtom), 'cDoubledAtom')
+const cQuadrupledAtom = atom((ctx) => 2 * ctx.spy(cDoubledAtom), 'cQuadrupledAtom')
+```
+
+### Some notes about atoms
+1. Atoms store things by reference. Just like props and useState in react.
+2. Computed atoms should be pure functions. You can't know how many times they will be called. But reatom will try to keep call count as small as possible.
```
Comment (artalar) [inline docs/src/content/docs/getting-started/learning.md]:
```suggestion
2. Computed atoms should be pure functions to archive the correct order of all computations
```

File: docs/src/content/docs/getting-started/learning.md [hunk 11]
```diff
@@ -0,0 +1,177 @@
...
+```typescript
+const someAtom = atom('Hello', 'someAtom')
+someAtom(ctx, (prev) => prev + ', atom')
+// someAtom -> 'Hello, atom'
+someAtom(ctx, (prev) => prev + ', atom')
+// someAtom -> 'Hello, atom, atom'
+```
+5. You can use previous state in computed atoms but you have to define types manually:
+```typescript
+const accAtom = atom<number>((ctx, prev = 0) => prev + 1, 'accAtom')
```
Comment (artalar) [inline docs/src/content/docs/getting-started/learning.md]: Very weird example :)
```suggestion
import { atom, isDeepEqual } from '@reatom/framework'

const listAtom = atom<List>([], 'listAtom')
const filteredListAtom = atom((ctx, state = []) => {
  const newState = ctx.spy(listAtom).filter(predicate)
  return isDeepEqual(state, newState) ? state : newState
}, 'filteredListAtom')
```

File: docs/src/content/docs/getting-started/learning.md [hunk 12]
```diff
@@ -0,0 +1,177 @@
...
+console.log(ctx1.get(someAtom))
+// logs: 1
+
+someExternalVar = 100
+
+// check using ctx1 and ctx2
+console.log(ctx1.get(someAtom), ctx2.get(someAtom))
+// logs: 1, 100
+// because ctx1 cached 1 and ctx2 was only read when value changed
+```
```
Comment (artalar) [inline docs/src/content/docs/getting-started/learning.md]:
```suggestion
```

To fix this, simply wrap the value in atom and spy, and change it within the relative context.
```

File: docs/src/content/docs/getting-started/debugging.md [hunk 13]
```diff
@@ -0,0 +1,69 @@
+---
+title: Debugging
+description: Explaining how to install and use debug tools
+sidebar:
+  order: 3
+---
+
```
Comment (artalar) [inline docs/src/content/docs/getting-started/debugging.md]:
```suggestion
<!-- TODO add `anAtom.onChange(console.log)` docs above -->
```

File: docs/src/content/docs/getting-started/debugging.md [hunk 14]
```diff
@@ -0,0 +1,69 @@
+---
```
Comment (artalar) [inline docs/src/content/docs/getting-started/debugging.md:1]: Please, fill this page with the original content (https://reatom-git-structured-docs-artalar.vercel.app/getting-started/debugging/) or mix it together

File: docs/astro.config.mjs [hunk 15]
```diff
@@ -25,45 +25,34 @@ export default defineConfig({
...
-          label: 'Guides',
+          label: 'Getting Started',
           autogenerate: {
-            directory: 'guides',
+            directory: 'getting-started',
           },
         },
         {
-          label: 'General',
+          label: 'Recipes',
```
Comment (artalar) [inline docs/astro.config.mjs:34]: Please, hide this whole section from the sidebar, as it not ready yet.

**Review by artalar — CHANGES_REQUESTED** [review]

File: docs/src/content/docs/getting-started/try-online.md (hunk 1 above)
Comment (Akiyamka) [reply docs/src/content/docs/getting-started/try-online.md:12]: How to install locally described with details in setup section

**Review by artalar — DISMISSED** [review]

**Pull request merged**